aws-datastore/src/androidTest/java/com/amplifyframework/datastore/storage/sqlite/SQLiteStorageAdapterClearTest.java

```diff
@@ -0,0 +1,217 @@
+        assertRecordIsInDb(blogger1);
+        //Verify observer is still alive
+        assertFalse(subscriberDisposableRef.get().isDisposed());
+        assertObserverReceivedRecord(blogger1);
+
+        adapter.clear();
+        //Make sure file was deleted and re-created
+        assertEquals(1, fileObserver.createFileEventCount);
+        assertEquals(1, fileObserver.deleteFileEventCount);
+        assertDbFileExists();
+        //Verify observer is still alive
+        assertFalse(subscriberDisposableRef.get().isDisposed());
+
+        //Make sure the new file is writable
+        adapter.save(blogger2);
+        //Check the new record is in the database
+        //and the old record is not.
+        assertRecordIsInDb(blogger2);
+        assertRecordIsNotInDb(blogger1);
+        assertObserverReceivedRecord(blogger2);
+        //Terminate the adapter
+        adapter.terminate();
+        //Verify observer was disposed.
+        assertTrue(subscriberDisposableRef.get().isDisposed());
+    }
+
+    private BlogOwner createBlogger(String name) throws DataStoreException {
+        return BlogOwner.builder()
+            .name(name)
+            .build();
+    }
+
+    private void assertObserverReceivedRecord(BlogOwner blogger) {
+        for (StorageItemChange<? extends Model> owner : observer.values()) {
+            if (BlogOwner.class.isAssignableFrom(owner.itemClass()) &&
+                blogger.getName().equals(((BlogOwner) owner.item()).getName())) {
+                return;
+            }
+        }
+        fail("Could not find " + blogger + " in event observer.");
+    }
+
+    private <T extends Model> void assertRecordIsInDb(T item) throws DataStoreException {
+        List<? extends Model> results = adapter.query(item.getClass(), Where.id(item.getId()));
+        assertEquals(1, results.size());
+        assertEquals(item, results.get(0));
+    }
+
+    private <T extends Model> void assertRecordIsNotInDb(T item) throws DataStoreException {
+        List<? extends Model> results = adapter.query(item.getClass(), Where.id(item.getId()));
+        assertNotNull(results);
+        assertEquals(0, results.size());
+    }
+
+    private void assertDbFileExists() {
+        assertTrue(context.getDatabasePath(DATABASE_NAME).exists());
+    }
+
+    private static final class TestFileObserver extends FileObserver {
+        private int createFileEventCount;
+        private int deleteFileEventCount;
+        /**
+         * Equivalent to FileObserver(file, FileObserver.ALL_EVENTS).
+         *
+         * @param path Directory to watch
+         */
+
+        @SuppressWarnings("deprecation")
+        TestFileObserver(@NonNull String path) {
+            super(path, FileObserver.CREATE | FileObserver.DELETE | FileObserver.DELETE_SELF);
+
+            this.deleteFileEventCount = 0;
+            this.createFileEventCount = 0;
+        }
+
+        /**
+         * The event handler, which must be implemented by subclasses.
+         *
```

Comment:
I believe the best practice is _not_ to provide Javadocs for `@Override` methods, since it will be mostly duplicative of the content of the interface/base class being overriden.

The [`JavadocMethod`](https://checkstyle.sourceforge.io/config_javadoc.html#JavadocMethod) checkstyle has this a default setting:

> Javadoc is not required on a method that is tagged with the @Override annotation.

In a few place in this code base, we have the `@inheritDoc` Javadoc directive on `@Override`n methods.

A guy on [StackOverflow](https://stackoverflow.com/a/3062697/695787) goes on:

> I studied this problem during my PhD and found that in general folks will never be aware of the extra information in the overriding version if they are invoking through a supertype.

Holy Cow. Talk about "Diving Deep!"